### ISSUE TITLE
[Mac] Issue in PixelLayout.Move

### ISSUE THREAD

**Issue by msasso69** [picoe/Eto#925]
Hi All,

I noticed a problem using PixelLayout.Move on Mac.

Unfortunately I did not manage to reproduce the issue in a simple snippet of code but, even if I already have a valid workaround for my code, I decided to report it anyway because the information that I can provide might be enough to somebody to diagnose and fix it for the benefit of the community ;)

Let me first describe what I'm doing and what happens: basically I have a PixelLayout where I add two controls, the first contains my application UI and it is sized to the the whole available client size, the second contains a progress bar that I show/hide for log running operations and that I position with the PixelLayout.Move method at the center of the first control. In order to correctly position the progress bar I re-position it at every PixelLayout.SizeChanged event.
On Wpf everything works fine but on Mac as soon as I call the PixelLayout.Move my code loops inside the following code stack until I get a stack overflow exception:

	VisualSEO.EtoFormGui.Forms.ViewTabularFormView.AnonymousMethod__2E(VisualSEO.EtoFormGui.Forms.ViewTabularFormView sender, System.EventArgs e) in /Users/utonto/vssrepo6/EtoFormGui/Forms/ViewTabularFormView.eto.cs:100
	Eto.PropertyStore.TriggerEvent<System.EventArgs>(string key, VisualSEO.EtoFormGui.Forms.ViewTabularFormView sender, System.EventArgs args) in /Users/utonto/Eto/Source/Eto/PropertyStore.cs:234
	Eto.Forms.Control.OnSizeChanged(System.EventArgs e) in /Users/utonto/Eto/Source/Eto/Forms/Controls/Control.cs:128
	Eto.Forms.Control.Callback.OnSizeChanged(VisualSEO.EtoFormGui.Forms.ViewTabularFormView widget, System.EventArgs e) in /Users/utonto/Eto/Source/Eto/Forms/Controls/Control.cs:1334
	Eto.Mac.Forms.MacView<MonoMac.AppKit.NSView,Eto.Forms.Panel,Eto.Forms.Control.ICallback>.SetFrameSizeAction(System.IntPtr sender, System.IntPtr sel, MonoMac.CoreGraphics.CGSize size) in /Users/utonto/Eto/Source/Eto.Mac/Forms/MacView.cs:374
	MonoMac.ObjCRuntime.Messaging.void_objc_msgSendSuper_CGRect() in 
	MonoMac.AppKit.NSView.set_Frame(MonoMac.CoreGraphics.CGRect value) in 
	Eto.Mac.Forms.Controls.SplitterHandler.ResizeSubviews(MonoMac.CoreGraphics.CGSize oldSize2) in /Users/utonto/Eto/Source/Eto.Mac/Forms/Controls/SplitterHandler.cs:261
	Eto.Mac.Forms.Controls.SplitterHandler.EtoSplitViewDelegate.Resize(Eto.Mac.Forms.Controls.SplitterHandler.EtoSplitView splitView, MonoMac.CoreGraphics.CGSize oldSize) in /Users/utonto/Eto/Source/Eto.Mac/Forms/Controls/SplitterHandler.cs:274
	object.[Eto.Mac.Forms.Controls.SplitterHandler+EtoSplitViewDelegate:Void Resize(MonoMac.AppKit.NSSplitView, MonoMac.CoreGraphics.CGSize)](Eto.Mac.Forms.Controls.SplitterHandler.EtoSplitViewDelegate arg0, System.IntPtr arg1, Eto.Mac.Forms.Controls.SplitterHandler.EtoSplitView arg2, MonoMac.CoreGraphics.CGSize arg3) in 
	MonoMac.ObjCRuntime.Messaging.void_objc_msgSendSuper_CGSize() in 
	MonoMac.AppKit.NSView.ResizeSubviewsWithOldSize(MonoMac.CoreGraphics.CGSize oldSize) in 
	Eto.Mac.Forms.Controls.SplitterHandler.UpdatePosition() in /Users/utonto/Eto/Source/Eto.Mac/Forms/Controls/SplitterHandler.cs:629
	Eto.Mac.Forms.Controls.SplitterHandler.LayoutParent(bool updateSize) in /Users/utonto/Eto/Source/Eto.Mac/Forms/Controls/SplitterHandler.cs:40
	Eto.Mac.Forms.MacContainer<MonoMac.AppKit.NSView,Eto.Forms.Panel,Eto.Forms.Control.ICallback>.LayoutParent(bool updateSize) in /Users/utonto/Eto/Source/Eto.Mac/Forms/MacContainer.cs:146
	Eto.Mac.Forms.MacContainer<MonoMac.AppKit.NSView,Eto.Forms.PixelLayout,Eto.Forms.Control.ICallback>.LayoutParent(bool updateSize) in /Users/utonto/Eto/Source/Eto.Mac/Forms/MacContainer.cs:146
	Eto.Mac.Forms.PixelLayoutHandler.Move(VisualSEO.EtoFormGui.Controls.TextProgressBarView child, int x, int y) in /Users/utonto/Eto/Source/Eto.Mac/Forms/PixelLayoutHandler.cs:137
	Eto.Forms.PixelLayout.Move(VisualSEO.EtoFormGui.Controls.TextProgressBarView control, int x, int y) in /Users/utonto/Eto/Source/Eto/Forms/Layout/PixelLayout.cs:106
	Eto.Forms.PixelLayout.Move(VisualSEO.EtoFormGui.Controls.TextProgressBarView control, Eto.Drawing.Point location) in /Users/utonto/Eto/Source/Eto/Forms/Layout/PixelLayout.cs:116
	VisualSEO.EtoFormGui.Forms.ViewTabularFormView.AnonymousMethod__2E(VisualSEO.EtoFormGui.Forms.ViewTabularFormView sender, System.EventArgs e) in /Users/utonto/vssrepo6/EtoFormGui/Forms/ViewTabularFormView.eto.cs:100

Just to make you understand the above code stack keep in mind that the PixelLayout is in the ViewTabularFormView and the progress bar is contained in the TextProgessBarView control. Notice that the Splitter that gets resized is not in the control that contains the progress bar rather in the control that I first add to the PixelLayout and that basically contains my UI in background.

Basically it looks like moving the progress (the overlaying control) triggers a resize of a control inside the control that I'm not moving (the background control) in the PixelLayout, but I have no explanation why it happens.

Hope this helps

## Specifications

  - Version: latest develop branch
  - Platform(s): Mac

**Comment by cwensley** [comment]
Thanks for submitting the issue.  For now, you may be able to get around this by only calling Move() when the control actually needs to move.. unless it's changing all the time?

**Comment by msasso69** [comment]
Thanks for your comment,
I haven't investigated any further on this issue since I already have a valid workaround for it (and other issues to work on that are hurting me more ;), but I sense it is related to #939 and how I size my controls.
I will check again with the latest "develop" code and add an update here if there are any changes.
:)

**Comment by msasso69** [comment]
I confirm the the issue is fixed by #1112

**Comment by cwensley** [comment]
@msasso69 awesome, thanks for following up!
